gcloud_dataproc/utils/seqr_api.py

```diff
@@ -1,94 +1,59 @@
-        # download fam file for this project
-        seqr_login_url = os.path.join(seqr_url, "login")
-        seqr_individual_api_url = os.path.join(seqr_url,
-                                               "api/project/{0}/export_project_individuals?file_format=tsv".format(project_guid))
+    s = requests.Session()
+    response = s.get(seqr_login_url)    # get CSRF cookies
+    response.raise_for_status()
 
-        s = requests.Session()
-        response = s.get(seqr_login_url)    # get CSRF cookies
-        response.raise_for_status()
+    csrf_token = s.cookies.get('csrftoken')
+    data = {
+        'username_or_email': seqr_username,
+        'password': seqr_password,
+        'csrfmiddlewaretoken': csrf_token,
+    }
+    response = s.post(seqr_login_url, data=data)    # submit login
 
-        csrf_token = s.cookies.get('csrftoken')
-        data = {
-            'username_or_email': seqr_username,
-            'password': seqr_password,
-            'csrfmiddlewaretoken': csrf_token,
-        }
-        response = s.post(seqr_login_url, data=data)    # submit login
+    response.raise_for_status()
 
-        response.raise_for_status()
+    response = s.get(seqr_individual_api_url.format(**locals()))   # get json from API
+    response.raise_for_status()
 
-        response = s.get(seqr_individual_api_url.format(**locals()))   # get json from API
-        response.raise_for_status()
+    content = str(response.content)
 
-        content = str(response.content)
-
-        fam_file_rows = []
-        subset_samples_rows = []
-        for line in content.split("\n"):
-            if not line or line.startswith("#"):
-                continue
-            if not fam_file_rows and "affected" in line.lower() and "indiv" in line.lower() and "family" in line.lower():
-                continue  # skip header row
-
-            fields = line.split("\t")
-            fam_file_rows.append(fields[0:6])
-            if len(fields) > 1:
-                subset_samples_rows.append(fields[1])
-
-        # upload fam file to same place as VCF
-        if not is_fam_file_specified:
-            with open(fam_filename, "w") as f:
-                f.write("\n".join(["\t".join(row) for row in fam_file_rows]))
-
-            run("gsutil cp %(fam_filename)s %(fam_file_gcloud_path)s" % locals())
-
-            command_line_args.extend(["--fam-file", fam_file_gcloud_path])
+    fam_file_rows = []
+    subset_samples_rows = []
+    for line in content.split("\n"):
+        if not line or line.startswith("#"):
+            continue
+        if not fam_file_rows and "affected" in line.lower() and "indiv" in line.lower() and "family" in line.lower():
+            continue  # skip header row
 
-        # upload subset-samples to same place as VCF
-        if not is_subset_samples_file_specified:
-            with open(subset_samples_filename, "w") as f:
-                f.write("\n".join([row for row in subset_samples_rows]))
+        fields = line.split("\t")
```

Comment:
I think this would be more resilient and easier to read if instead of using a file download we use the `/api/project/<project_guid>/details` endpoint and then get the data from the `individualsByGuid` key of the response. That way we are actually getting the fields we want explicitly instead of relying on the columns of this download to never change

Reply:
The current version isn't ideal but the tsv is slightly less brittle than it looks because it's following .fam file format. I agree json is much better, but I'd vote for staying with the current api and just adding a format=json option - that way it's only retrieving individuals and isn't affected by changes to the project page.